libraries/Microsoft.Bot.Connector/Authentication/AppCredentials.cs

```diff
@@ -100,6 +100,14 @@ public string ChannelAuthTenant
         /// </value>
         public virtual string OAuthEndpoint => string.Format(CultureInfo.InvariantCulture, AuthenticationConstants.ToChannelFromBotLoginUrlTemplate, ChannelAuthTenant);
 
+        /// <summary>
+        /// Gets a value indicating whether to validate the Authority.
+        /// </summary>
+        /// <value>
+        /// The ValidateAuthority value to use.
+        /// </value>
+        public virtual bool ValidateAuthority => true;
```

Comment:
I'm curious, what are the cases where we or developers wouldn't want to validate the authority?

Reply:
Might also be worth explaining in the docs.

Reply:
We only want to validate the authority for Public and Gov clouds. Other private or sovereign clouds need this to be false. (This change comes from @willportnoy.)

Reply:
Fixed list of authorities here:

https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/blob/bc9b6cc15ad271c1f4f9f1ccdbad6360b826080f/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Instance/InstanceDiscovery.cs#L69